Makefile

```diff
@@ -52,7 +52,7 @@ LDFLAGS += -extldflags "-static"
 endif
 
 .PHONY: all
-all: build test
+all: build test check
```

Comment:
`make check` looks heavily parallelizable (with a small `Go` wrapper: spawn all in parallel, wait and return those stdouts/errs for exit code `!= 0`) if we think it's worth it.
And for `make test`, since all of our tests can theoretically run in parallel, we could play with http://golang.org/pkg/testing/#T.Parallel and another optimization is running all package-level tests in parallel (maybe could use the same wrapper as above).